### PR TITLE
Add option to make `_validAgentAndTime` 3.x compatible

### LINKED ISSUE #10563: CakePHP 2.x // 3.x Session.time inconsistent
This is a (multiple allowed):

* [x] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.4.1
* Platform and Target: nginx / apache, mysql

### What you did
Tried to migrate old CakePHP 2 project step by step to version 3.x, i put a nginx proxy before both versions and route everything to the old version. If a new controller is able to handle the user needs i route with nginx reverse proxy /converted_controller_name to the CakePHP 3 project.

### What happened
Session closed, coming back from CakePHP 3 back to CakePHP 2 (browser back button or existing link) because of time expired.

#### CakePHP 2 Session Validation

// lib/Cake/Model/Datasource/CakeSession.php

	protected static function _validAgentAndTime() {
		$userAgent = static::read('Config.userAgent');
		$time = static::read('Config.time');
		$validAgent = (
			Configure::read('Session.checkAgent') === false ||
			isset($userAgent) && static::$_userAgent === $userAgent
		);
		return ($validAgent && static::$time <= $time);
	}

#### CakePHP 3 Session Timeout Check

// src/Network/Session.php

    protected function _timedOut()
    {
        $time = $this->read('Config.time');
        $result = false;

        $checkTime = $time !== null && $this->_lifetime > 0;
        if ($checkTime && (time() - $time > $this->_lifetime)) {
            $result = true;
        }

        $this->write('Config.time', time());

        return $result;
    }

EXPLAIN WHAT IS TO BE EXPECTED, HERE.

The logout is caused because CakePHP 2 and 3 are using Config.time in a different way, once with the last time the session was written and once the time when the session expires. Would be helpful to ether update the name to something different or use it in the same way. Otherwise its hard to share a session between two versions of the same framework.

### PR DESCRIPTION
**Pull request by tersmitten**
fixes #10563

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]
Looks good to me (and BC).

File: lib/Cake/Model/Datasource/CakeSession.php [hunk 1]
```diff
@@ -527,6 +537,10 @@ protected static function _configureSession() {
 		if (isset($sessionConfig['timeout']) && !isset($sessionConfig['cookieTimeout'])) {
 			$sessionConfig['cookieTimeout'] = $sessionConfig['timeout'];
 		}
+		if (isset($sessionConfig['useForwardsCompatibleTimeout']) && $sessionConfig['useForwardsCompatibleTimeout']) {
+				static::$_useForwardsCompatibleTimeout = true;
```
Comment (markstory) [inline lib/Cake/Model/Datasource/CakeSession.php]: This looks like it has an extra layer of indentation.

Comment (tersmitten) [reply lib/Cake/Model/Datasource/CakeSession.php]: I'll have a look

Comment (tersmitten) [reply lib/Cake/Model/Datasource/CakeSession.php]: You were right, you hawk :-)

**Comment by tersmitten** [comment]
Thanks

**Pull request merged**